Sources/EduIDExpansion/Flows/Scanning/VerifyScanResultViewController.swift

```diff
@@ -0,0 +1,120 @@
+import UIKit
+import TiqrCoreObjC
+import TinyConstraints
+
+class VerifyScanResultViewController: BaseViewController {
+    
+    let viewModel: ScanViewModel
+    weak var delegate: VerifyScanResultViewControllerDelegate?
+
+    //MARK: - init
+    init(viewModel: ScanViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    //MARK: - lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        screenType = .verifyLoginScreen
+        setupUI()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        screenType.configureNavigationItem(item: navigationItem)
+    }
+    //MARK: - setupUI
+    func setupUI() {
+        // - top poster label
+        let posterParent = UIView()
+        let posterLabel = UILabel.posterTextLabelBicolor(text: "Request to login", primary: "Request to login")
+        posterParent.addSubview(posterLabel)
+        posterLabel.edges(to: posterParent)
+        
+        let upperspace = UIView()
+        
+        // - text in middle with organisation name
+        let middlePosterParent = UIView()
+        var middlePosterLabel = UILabel()
+        
+        switch viewModel.challengeType {
+        case .enrollment:
+            middlePosterLabel = UILabel.requestLoginLabel(entityName: (viewModel.challenge as? EnrollmentChallenge)?.identityDisplayName ?? "", challengeType: viewModel.challengeType ?? .invalid)
+        case .authentication:
+            middlePosterLabel = UILabel.requestLoginLabel(entityName: (viewModel.challenge as? AuthenticationChallenge)?.serviceProviderDisplayName ?? "", challengeType: viewModel.challengeType ?? .invalid)
+        default:
```

Comment:
Avoid default cases if possible. That way if you later add a new case, the compiler tells you all the places you need to revisit.

Reply:
I resolved this